packages/styleguide/src/components/Preview/Preview.js

```diff
@@ -26,310 +26,267 @@ import Button from './../Button';
-            className="select-wrapper"
-            classNamePrefix="select"
-            isSearchable={false}
-            isClearable={false}
-            value={previewBackground}
-            placeholder={previewBackground.value}
-            onChange={this.handlePreviewBackground}
-            options={bgColorsOptions}
-            styles={colourStyles}
-          />
-        );
-      }
-    }
+  const classes = cx(CLASS_ROOT, className);
+  const wrapperClasses = cx('preview-wrapper', {
+    'is-fullscreen': isFullscreen,
+  });
+
+  const colourStyles = {
+    option: (styles, { data, isActive }) => {
+      const color = chroma(data.value);
+      return {
+        ...styles,
+        backgroundColor: isActive ? 'black' : data.value,
+        color: chroma.contrast(color, 'white') > 2 ? 'white' : 'black',
+        cursor: 'pointer',
+      };
+    },
+    container: () => {},
+    control: () => {},
+    valueContainer: () => {},
+    singleValue: () => {},
+    indicatorSeparator: () => {},
+  };
 
-    if (hasCodePreview) {
-      actions.push(
-        <Button onClick={this.handleToggleCode}>
-          <Icon name="code" fill={bgThemeColors.accent || '#000'} />
-          {this.state.isCodeShown ? 'HIDE CODE' : 'SHOW CODE'}
-        </Button>
-      );
-    }
+  const actions = [];
 
-    if (isInteractive) {
-      actions.push(
-        <Button onClick={this.handleToggleInteract}>
-          <Icon name="code" fill={bgThemeColors.accent || '#000'} />
-          {this.state.showInteract ? 'HIDE INTERACTIVE' : 'SHOW INTERACTIVE'}
-        </Button>
-      );
-    }
+  if (bgTheme && bgThemeColors) {
+    const bgColorsOptions = getBackgroundsAsArray(
+      bgThemeColors,
+      bgThemeExcludedColors
+    );
 
-    if (enableFullscreen) {
+    if (bgColorsOptions.length) {
       actions.push(
-        <Button onClick={this.handleToggleFullscreen}>
-          <Icon
-            name={this.state.isFullscreen ? 'fullscreen-exit' : 'fullscreen'}
-          />
-          {this.state.isFullscreen ? 'MINIMIZE' : 'FULLSCREEN'}
-        </Button>
+        <StyledSelect
+          name="background-select"
+          className="select-wrapper"
+          classNamePrefix="select"
+          isSearchable={false}
+          isClearable={false}
+          value={previewBackground}
+          placeholder={previewBackground.value}
+          onChange={handlePreviewBackground}
+          options={bgColorsOptions}
+          styles={colourStyles}
+        />
       );
     }
+  }
 
-    const renderAsFunctionContext = {
-      bgTheme: previewBackground.label,
-      bgThemeValue: previewBackground.value,
-    };
-
-    const childrenToRender =
-      typeof children === 'function'
-        ? children(renderAsFunctionContext)
-        : children;
-
-    const toReneder = childrenToRender || (
-      // eslint-disable-next-line react/no-danger
-      <div dangerouslySetInnerHTML={{ __html: html }} />
+  // TODO zjednotit
```

Comment:
Zjednotit co? :)

Reply:
su tu tri velmi podobne if statementy na to sme zabudli :D

Reply:
a co je s nimi zle? :)

Reply:
nič :D ale opakovanie troch identických vecí hneď za sebou mi rozsvieťuje kontrolku v hlave že niečo nie je programátorsky správne :) ale kým sú za sebou tak je to aspoň prehľadné. keby boli v troch rôznych fileoch už by to bolo menej prehľadné pri refaktoringu.